etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/custom/UpdateUniversalServiceIdentifier.java

```diff
@@ -23,33 +23,40 @@ public class UpdateUniversalServiceIdentifier implements CustomFhirTransformatio
     public static final String ALTERNATE_ID_NAME = "alternateId";
 
     @Override
-    public void transform(FhirResource<?> resource, Map<String, String> args) {
+    public void transform(FhirResource<?> resource, Map<String, Object> args) {
         Bundle bundle = (Bundle) resource.getUnderlyingResource();
         var serviceRequests = HapiHelper.resourcesInBundle(bundle, ServiceRequest.class);
 
         serviceRequests.forEach(
                 it -> {
                     var allCodings = it.getCode().getCoding();
                     var codingSystemContainer =
-                            getCodingSystemContainer(allCodings, args.get(CHECK_VALUE_NAME));
+                            getCodingSystemContainer(
+                                    allCodings, castToString(args.get(CHECK_VALUE_NAME)));
 
                     if (codingSystemContainer == null) {
                         // we're only interested in coding that matches the checkValue argument
                         return;
                     }
 
                     // check for the coding system label and create or override it
-                    updateCodingSystemLabel(codingSystemContainer, args.get(CODING_SYSTEM_NAME));
+                    updateCodingSystemLabel(
+                            codingSystemContainer, castToString(args.get(CODING_SYSTEM_NAME)));
 
                     // the alt id is stored on a separate coding object, so we need to filter
                     // for it
-                    String alternateId = args.get(ALTERNATE_ID_NAME);
+                    String alternateId = castToString(args.get(ALTERNATE_ID_NAME));
                     if (alternateId != null) {
                         updateAlternateCodingId(allCodings, alternateId);
                     }
                 });
     }
 
+    // Returns null if it is not a String
+    private String castToString(Object obj) {
+        return obj instanceof String ? (String) obj : null;
```

Comment:
Again, do we want this to skip or pass an empty string ?

Reply:
Since we control the values passed in args, I think it's safe to assume they will be there and have the correct type. If that's not the case, we need to know so we can fix it, so it's better to let it fail